### ISSUE TITLE
gamepads conflicts

### ISSUE THREAD

**Issue by ghost** [ares-emulator/ares#11]
Hello good afternoon.

I would like to report: When configuring gamepads from different consoles, the settings conflicts, not being separate one setting for each console.

Thank you.

**Comment by LukeUsher** [comment]
This is intentional by design: ares is designed to use a single 'virtual' gamepad that gets mapped to each system, similar to how RetroArch and other multi-system emulators work.

The individual system mappings are shown purely for convenience.

**Comment by ghost** [comment]
Understand. Thank you for clarifying this question, I really appreciate the work. Congratulations to everyone involved.

**Comment by FitzRoyX** [comment]
Out of curiosity, which 2 systems were you trying to map when you felt there was a conflict?

**Comment by ghost** [comment]
@FitzRoyX I started by mapping the Famicom, and next i mapped the Super Famicom and Nintendo 64. Only after doing this, I realized that the systems share the same mapping configuration. I realized that all the changes i made had an change on the "Virtual Gamepads". I thought each tab was a different setting, just like it is in RetroArch, but now Luke explained to me that it's different. But that's ok, I'll keep tweaking every time I start the game when I need to, that's not a big issue. I really appreciate your work, I like this system because it creates less files in Windows compared to RetroArch. For my type of use it is ideal and sensational. Thank you very much!

**Comment by RokkumanX** [comment]
If it helps, I use different folders for different systems, not the best way to solve the problem but it gets the job done.

For example I use Game Boy, Game Boy Color, Nintendo DS, Nintendo Virtual Boy and Sega Genesis in RetroArch and they each have their own folders with their own configurations.

It does however take some space using 5 instances of RetroArch, but I have tailor made customizations for each and every system so having them in one wouldn't work that good.

As for ares I only use it for Super Famicom games so that isn't much of a big deal for me.

TL;DR

Use separate ares folders for each of the systems you want to play, that way you can have customized configurations for those systems.

**Comment by ghost** [comment]
@RokkumanX Thanks for the reply this is a good tip, i will do it. RetroArch is excellent too, i really admire the work done on it, but sometimes i get annoyed that it creates approximately 10000 files on Windows. This is because it is complete and with a great and lot configurations options and has a lot themes. 
I decided that i'm going to migrate to ares. So I'm going to make this method, creating separate folders and then the settings will be separate for each system.
I will also do this in melonds standalone, because some games i like to play with different control profiles.
Thanks for the answer. Have a nice day.👍😀

**Comment by RokkumanX** [comment]
@SoloBackup 

Good to hear!

Off-topic:

Here's a nice other little tip for you when it comes to RetroArch, see when you extract RetroArch it's correct that it contains thousands of files but if you know what to do you could slim it down considerably.

My Sega Genesis RetroArch setup for example contains only the minimal needed assets, 378 files and 47 folders.
It only contains the necessary files for Sega Genesis and RetroArch to run.

Most emulators contains files not needed and you can get away with much.

**Comment by ghost** [comment]
@RokkumanX Thank you, so i'll try to delete the folders one by one and see which ones don't disturb the progress of the emulator. I will do this calmly and slowly so as not to make a mistake. I will also keep my attention on this project ares is very good and has some differences that I like. Congratulations to everyone involved.

**Comment by RokkumanX** [comment]
@SoloBackup 

I really like your enthusiasm, but ask yourself this:
What is it that RetroArch contains the most junk of?

Themes and shaders mostly!

I don't need anything else than the default theme so that means all themes except the default goes, +/- a few assets related to the theme of course.

And for Sega Genesis for example, I don't need more than 1 or 2 of the hundreds upon hundreds of shaders included with RetroArch.

Then I just toss things related to other cores and systems that isn't the one I'm trying to emulate.

Also if an emulator have the opportunity to change language and there are language files, I usually whack them too, I only use English and nothing else.

That's more or like it, .exe and .dll and other important files are a big no no so those I stay away from.

Emulators/frontends like RetroArch needs to be cleaned because it contains a lot of junk you don't need.

**Comment by ghost** [comment]
@RokkumanX 
Yesterday i spent the day doing the cleaning, and the result was incredible, before there were around 10000 files, and now there are around 463 files, this makes it much easier when copying to another directory. I was so excited that i did it in Dolphin too, which also had a nice effect, deleted the themes and some configuration files for games i don't play, and it got a lot cleaner too.
It was very organized now, I even made a backup to not lose this scheme. This tip was gold. Thank you buddy!
